src/sentry/mediators/sentry_apps/updater.py

```diff
@@ -77,12 +78,30 @@ def _update_events(self):
         from sentry.mediators.service_hooks.creator import expand_events
 
         self.sentry_app.events = expand_events(self.events)
-        self._update_service_hook_events()
 
-    def _update_service_hook_events(self):
+    def _update_service_hooks(self):
         hooks = ServiceHook.objects.filter(application=self.sentry_app.application)
-        for hook in hooks:
-            service_hooks.Updater.run(service_hook=hook, events=self.events)
+        if len(hooks) > 0:
```

Comment:
@MeredithAnya 
1. We need some check to make sure that there were not already service hooks already before creating a new one. Though we could refactor this to move the check to the bottom condition:
```python
for hook in hooks:
    if self.webhook_url:
        service_hooks.Updater.run(
            service_hook=hook, events=self.sentry_app.events, url=self.webhook_url
        )
    else:
         service_hooks.Destroyer.run(service_hook=hook)

if self.webhook_url and len(hook) === 0 && self.sentry_app.is_internal:
    installation = SentryAppInstallation.objects.get(sentry_app_id=self.sentry_app.id)
    service_hooks.Creator.run(
        application=self.sentry_app.application,
        actor=installation,
        projects=[],
        organization=self.sentry_app.owner,
        events=self.sentry_app.events,
        url=self.webhook_url,
    )
```

2. Good catch about the `self.webhook_url` catch. If they don't pass that attribute, I think it would delete the hooks. Let me add a unit test for this case.